src/node_api.cc

```diff
@@ -0,0 +1,2478 @@
+  } while (0)
+
+#define CHECK_NEW_FROM_UTF8(isolate, result, str) \
+  CHECK_NEW_FROM_UTF8_LEN((isolate), (result), (str), -1)
+
+#define GET_RETURN_STATUS()        \
+  (!try_catch.HasCaught() ? napi_ok \
+                         : napi_set_last_error(napi_pending_exception))
+
+// Static last error returned from napi_get_last_error_info
+napi_extended_error_info static_last_error = { nullptr, nullptr, 0, napi_ok };
+
+// Warning: Keep in-sync with napi_status enum
+const char* error_messages[] = {nullptr,
+                                "Invalid pointer passed as argument",
+                                "An object was expected",
+                                "A string was expected",
+                                "A function was expected",
+                                "A number was expected",
+                                "A boolean was expected",
+                                "An array was expected",
+                                "Unknown failure",
+                                "An exception is pending"};
+
+void napi_clear_last_error() {
+  static_last_error.error_code = napi_ok;
+
+  // TODO(boingoing): Should this be a callback?
+  static_last_error.engine_error_code = 0;
+  static_last_error.engine_reserved = nullptr;
+}
+
+const napi_extended_error_info* napi_get_last_error_info() {
+  static_assert(node::arraysize(error_messages) == napi_status_last,
+      "Count of error messages must match count of error values");
+  assert(static_last_error.error_code < napi_status_last);
+
+  // Wait until someone requests the last error information to fetch the error
+  // message string
+  static_last_error.error_message =
+      error_messages[static_last_error.error_code];
+
+  return &static_last_error;
+}
+
+napi_status napi_set_last_error(napi_status error_code,
+                                uint32_t engine_error_code = 0,
+                                void* engine_reserved = nullptr) {
+  static_last_error.error_code = error_code;
+  static_last_error.engine_error_code = engine_error_code;
+  static_last_error.engine_reserved = engine_reserved;
+
+  return error_code;
+}
+
+napi_status napi_create_function(napi_env env,
+                                 const char* utf8name,
+                                 napi_callback cb,
+                                 void* callback_data,
+                                 napi_value* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(result);
+
+  v8::Isolate* isolate = v8impl::V8IsolateFromJsEnv(env);
+  v8::Local<v8::Function> return_value;
+  v8::EscapableHandleScope scope(isolate);
+  v8::Local<v8::Object> cbdata =
+      v8impl::CreateFunctionCallbackData(env, cb, callback_data);
+
+  RETURN_STATUS_IF_FALSE(!cbdata.IsEmpty(), napi_generic_failure);
+
+  v8::Local<v8::FunctionTemplate> tpl = v8::FunctionTemplate::New(
+      isolate, v8impl::FunctionCallbackWrapper::Invoke, cbdata);
+
+  return_value = scope.Escape(tpl->GetFunction());
```

Comment:
@jasongin What I’ve done is to just cd into `out`, take the compile command that ran with a normal `make` and add `-DV8_IMMINENT_DEPRECATION_WARNINGS=1` manually… that worked well enough for me